api_test.go

```diff
@@ -564,3 +565,75 @@ func TestGetOAuthClients(t *testing.T) {
 
 	ApiSpecRegister = nil
 }
+
+func TestResetHandler(t *testing.T) {
+	uri := "/tyk/reload/"
+
+	ApiSpecRegister = make(map[string]*APISpec)
+
+	makeSampleAPI(t, apiTestDef)
+
+	recorder := httptest.NewRecorder()
+	params := make(url.Values)
+
+	req, err := http.NewRequest("GET", uri+params.Encode(), nil)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resetHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Fatal("Hot reload failed, response code was: ", recorder.Code)
+	}
+
+	if len(ApiSpecRegister) == 0 {
+		t.Fatal("Hot reload was triggered but no APIs were found.")
+	}
+}
+
+func TestGroupResetHandler(t *testing.T) {
+	signalChan := make(chan bool)
+	cacheStore := RedisClusterStorageManager{}
+	cacheStore.Connect()
+
+	go func(pubSubChannel string, store RedisClusterStorageManager, signalChan chan bool) {
+		store.StartPubSubHandler(pubSubChannel, func(message redis.Message) {
+			notif := Notification{}
+			if err := json.Unmarshal(message.Data, &notif); err != nil {
+				t.Fatal("Unmarshalling message body failed, malformed: ", err)
+			}
+			if notif.Command == NoticeGroupReload {
+				signalChan <- true
```

Comment:
If the condition isn't met, or this line doesn't execute for any other reason, the main goroutine will block forever. The test will use up all of the timeout. Whatever your purpose here is, there has to be a better way to do it.